test/Microsoft.ML.Benchmarks/BigramAndTrigramBenchMark.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.ML.Runtime.Tools;
+using Microsoft.ML.TestFramework;
+using System.IO;
+
+namespace Microsoft.ML.Benchmarks
+{
+    public class BigramAndTrigramBenchmark
+    {        
+        private static string s_dataPath_Wiki;
+        private static string s_output_Wiki;
+
+        [Benchmark]
+        public void Preceptron_CV()
+        {
+            string outPath = Path.Combine(s_output_Wiki, @"0.model.zip");
+            string cmd = @"CV k=5 data=" + s_dataPath_Wiki + " loader=TextLoader{quote=- sparse=- col=Label:R4:0 col=rev_id:TX:1 col=comment:TX:2 col=logged_in:BL:4 col=ns:TX:5 col=sample:TX:6 col=split:TX:7 col=year:R4:3 header=+} xf=Convert{col=logged_in type=R4} xf=CategoricalTransform{col=ns} xf=TextTransform{col=FeaturesText:comment wordExtractor=NGramExtractorTransform{ngram=2}} xf=Concat{col=Features:FeaturesText,logged_in,ns} tr=OVA{p=AveragedPerceptron{iter=10}} out={" + outPath + "}";
+            Maml.MainAll(cmd);
+        }
+
+        [Benchmark]
+        public void LightGBM_CV()
+        {
+            string outPath = Path.Combine(s_output_Wiki, @"1.model.zip");
+            string cmd = @"CV k=5 data=" + s_dataPath_Wiki + " loader=TextLoader{quote=- sparse=- col=Label:R4:0 col=rev_id:TX:1 col=comment:TX:2 col=logged_in:BL:4 col=ns:TX:5 col=sample:TX:6 col=split:TX:7 col=year:R4:3 header=+} xf=Convert{col=logged_in type=R4} xf=CategoricalTransform{col=ns} xf=TextTransform{col=FeaturesText:comment wordExtractor=NGramExtractorTransform{ngram=2}} xf=Concat{col=Features:FeaturesText,logged_in,ns} tr=LightGBMMulticlass{} out={" + outPath + "}";
+            Maml.MainAll(cmd);
+        }
+
+        [Benchmark]
+        public void WikiDox_Test()
+        {
+            string modelPath = Path.Combine(s_output_Wiki, @"0.model.fold000.zip");
+            string cmd = @"Test data=" + s_dataPath_Wiki + " in=" + modelPath;
+            Maml.MainAll(cmd);
```

Comment:
>Maml.MainAll(cmd); [](start = 12, length = 18)

don't know much about benchmark tests: do we need to track the time the run starts-ends or the framework does it for us?

Reply:
[Benchmark] attribute automatically does that for you. At the end of the run it gives you all the statistics like mean, median, memory used